### PR TITLE
Img vol return

### PR DESCRIPTION
**Pull request by garrettwrong**
Initial pass patches for `Image` done.  Working `Volume` now.

Patching these tests sort of making me wonder if we really want to do this... TBD, can discuss.

### CODE REVIEW

File: src/aspire/source/image.py [hunk 1]
```diff
@@ -380,7 +380,7 @@ def _apply_filters(
         for i, filt in enumerate(filters):
             idx_k = np.where(indices == i)[0]
             if len(idx_k) > 0:
-                im[idx_k] = Image(im[idx_k]).filter(filt).asnumpy()
+                im[idx_k] = im[idx_k].filter(filt).asnumpy()
```
Comment (j-c-c) [inline src/aspire/source/image.py:383]: I know it's not something you touched in this PR, but the logic above this in lines 371-378 seems off. If `im_orig` is not an `Image` we set `im = Image(im_orig)`, but then we reset to `im = im_orig.copy()`. Should line 376 just be `im_orig = Image(im_orig)`. Sorry if this is outside the scope of this PR.

File: src/aspire/volume/volume.py [hunk 2]
```diff
@@ -176,26 +181,43 @@ def __radd__(self, otherL):
...
+        Scalar division, follows numpy semantics.
+        """
+        if isinstance(other, Volume):
+            res = self.__class__(self._data / other.asnumpy())
+        else:
+            res = self.__class__(self._data / other)
+
+        return res
+
+    def __rtruedive__(self, otherL):
```
Comment (j-c-c) [inline src/aspire/volume/volume.py]: A meaningful typo: `_rtruediv_`

Should we put this under testing?

File: src/aspire/volume/volume.py [hunk 3]
```diff
@@ -176,26 +181,43 @@ def __radd__(self, otherL):
...
+        """
+        Right scalar division, follows numpy semantics.
+        """
+        return self / otherL
```
Comment (j-c-c) [inline src/aspire/volume/volume.py]: Shouldn't this be `otherL / self`?

**Review by j-c-c — CHANGES_REQUESTED** [review]
Looks good (and tedious)! Just a couple things.

File: src/aspire/source/image.py (hunk 1 above)
Comment (garrettwrong) [reply src/aspire/source/image.py:383]: at line 378, im_orig should be an Image.

We copy that in all cases.

`Image.copy()` will take an image and return an image with the data deep copied.  This is to not mutate the original image data.

If I'm still missing something, lmk.

File: src/aspire/volume/volume.py (hunk 2 above)
Comment (garrettwrong) [reply src/aspire/volume/volume.py]: lol, another good deed.  Yeah, I'll fix this along with the other remark then add a test, thanks!

File: src/aspire/volume/volume.py (hunk 3 above)
Comment (garrettwrong) [reply src/aspire/volume/volume.py]: sort of, that would be right divison, need to do something different here.

File: src/aspire/source/image.py (hunk 1 above)
Comment (j-c-c) [reply src/aspire/source/image.py:383]: Why should `im_orig` be an `Image` at line 378? If it was not supplied as an `Image` it is not being converted to an `Image` anywhere.

Comment (garrettwrong) [reply src/aspire/source/image.py:383]: L376?

Comment (garrettwrong) [reply src/aspire/source/image.py:383]: Oh i see, someone made a typo.

Comment (garrettwrong) [reply src/aspire/source/image.py:383]: 376 should be im_orig, I'll change it. Good catch, thanks!

**Review by j-c-c — DISMISSED** [review]
LGTM!

File: tests/test_FFBbasis2D.py [hunk 4]
```diff
@@ -113,19 +113,19 @@ def testRotate(self, basis):
...
+        y2 = basis.evaluate(v2).asnumpy()
+        y3 = basis.evaluate(v3).asnumpy()
+        y4 = basis.evaluate(v4).asnumpy()
 
         # Rotate 90
         assert np.allclose(y1[0], y2[0], atol=1e-5)
 
         # 2*pi Identity
-        assert np.allclose(x1[0], y3[0], atol=1e-5)
+        assert np.allclose(x1.asnumpy()[0], y3[0], atol=1e-5)
```
Comment (janden) [inline tests/test_FFBbasis2D.py]: If we implement the array interface, shouldn't `np.allclose(x1[0], y3[0])` work?

**Review by janden — CHANGES_REQUESTED** [review]
Frankly, I think this all looks very reasonable. Just one question.

File: tests/test_FFBbasis2D.py (hunk 4 above)
Comment (garrettwrong) [reply tests/test_FFBbasis2D.py]: indeed it does, fixed.

**Review by janden — APPROVED** [review]

**Pull request merged**